src/EvoSC.Common/Services/PlayerManagerService.cs

```diff
@@ -64,62 +64,21 @@ public async Task<IPlayer> CreatePlayerAsync(string accountId)
 
     public async Task<IOnlinePlayer> GetOnlinePlayerAsync(string accountId)
     {
-        var playerLogin = PlayerUtils.ConvertAccountIdToLogin(accountId);
-        // TODO: #74 Optimize Player State Fetching (https://github.com/EvoTM/EvoSC-sharp/issues/74)
-        var onlinePlayerInfo = await _server.Remote.GetPlayerInfoAsync(playerLogin);
-        var onlinePlayerDetails = await _server.Remote.GetDetailedPlayerInfoAsync(playerLogin);
-
-        if (onlinePlayerDetails == null || onlinePlayerInfo == null)
-        {
-            throw new PlayerNotFoundException(accountId, $"Cannot find online player: {accountId}");
-        }
-
-        var player = await GetOrCreatePlayerAsync(accountId);
+        var player = await _playerCache.GetOnlinePlayerCachedAsync(accountId);
 
         if (player == null)
         {
-            throw new PlayerNotFoundException(accountId);
+            throw new InvalidOperationException("Failed to get online player from cache.");
```

Comment:
Instead of throwing a new exception, just throw the one from inside GetOnlinePlayerCachedAsync. That way you get the same stacktrace and a single exception, not two.

Reply:
There are no duplicate exceptions here? I'm confused. The PlayerNotFoundException is also way more descriptive than InvalidOperationException.

Reply:
The two stacktraces are not even related.